server/src/test/java/io/druid/server/coordinator/CostBalancerStrategyTest.java

```diff
@@ -212,4 +211,58 @@ public void testIntervalCost() throws Exception
     // partial overlap [0, 3), [1, 2)
     Assert.assertEquals(1.534912, CostBalancerStrategy.intervalCost(3, 1, 2), 1e-6);
   }
+
+  @Test
+  @Ignore
+  public void testSpeedOfCostBalanceStrategy() throws InterruptedException
+  {
+    final int numServers = 70;
+    final int loaded = 200000;
+    final int adding = 10000;
+
+    List<DruidServer> servers = Lists.newArrayListWithCapacity(numServers);
+    for (int i = 1; i <= numServers; i++) {
+      servers.add(new DruidServer("server-" + i, "host-" + i, 1000000000000L, "hot", "hot", 1));
+    }
+
+    final LoadQueuePeon peon = new LoadQueuePeonTester();
+    final BalancerStrategy strategy = new CostBalancerStrategy(
+        MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(4)), 4
+    );
+
+    DateTime time = new DateTime("2014-01-01");
+    for (int i = 1; i <= loaded; i++) {
+      DataSegment segment = getSegment(i, "ds", new Interval(time, time = time.plusHours(1)));
+      servers.get(i % servers.size()).addDataSegment(segment.getIdentifier(), segment);
+    }
+
+    // should be materialized
+    final List<ServerHolder> holders = Lists.newArrayList(
+        Lists.transform(
+            servers,
+            new Function<DruidServer, ServerHolder>()
+            {
+              @Override
+              public ServerHolder apply(DruidServer input)
+              {
+                return new ServerHolder(input.toImmutableDruidServer(), peon);
+              }
+            }
+        )
+    );
+
+    System.out.println("started");
+    long start = System.currentTimeMillis();
+    long prev = System.currentTimeMillis();
+
+    for (int i = 1; i <= adding; i++) {
+      DataSegment segment = getSegment(i + loaded, "ds", new Interval(time, time = time.plusHours(1)));
+      strategy.findNewSegmentHomeReplicator(segment, holders);
+      if (i % 1000 == 0) {
+        System.out.println(i + " th.. took " + (System.currentTimeMillis() - prev) + " msec");
+        prev = System.currentTimeMillis();
+      }
+    }
+    System.out.println("Total " + (System.currentTimeMillis() - start) + " msec");
```

Comment:
the test is not for validation but for simple benchmarking. should we remove this?